feature-libs/product-configurator/rulebased/components/attribute/types/checkbox-list/configurator-attribute-checkbox-list.component.ts

```diff
@@ -1,69 +1,159 @@
+  attributeCheckBoxForms = new Array<FormControl>();
+
   @Input() group: string;
-  @Input() ownerKey: string;
 
-  @Output() selectionChange = new EventEmitter<ConfigFormUpdateEvent>();
+  // TODO(#11681): make quantityService a required dependency and remove deprecated constructor
+  /**
+   * default constructor
+   * @param {ConfiguratorStorefrontUtilsService} configUtilsService
+   * @param {ConfiguratorAttributeQuantityService} quantityService
+   */
+  constructor(
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
+    configUtilsService: ConfiguratorStorefrontUtilsService,
+    // eslint-disable-next-line @typescript-eslint/unified-signatures
+    quantityService: ConfiguratorAttributeQuantityService
+  );
+
+  /**
+   * @deprecated since 3.3
+   */
+  constructor(configUtilsService: ConfiguratorStorefrontUtilsService);
 
   constructor(
-    protected configUtilsService: ConfiguratorStorefrontUtilsService
+    protected configUtilsService: ConfiguratorStorefrontUtilsService,
+    protected quantityService?: ConfiguratorAttributeQuantityService
   ) {
-    super();
+    super(quantityService);
   }
 
-  attributeCheckBoxForms = new Array<FormControl>();
+  ngOnInit(): void {
+    const disabled = !this.allowZeroValueQuantity;
 
-  ngOnInit() {
     for (const value of this.attribute.values) {
       let attributeCheckBoxForm;
+
       if (value.selected === true) {
-        attributeCheckBoxForm = new FormControl(true);
+        attributeCheckBoxForm = new FormControl({
+          value: true,
+          disabled: disabled,
+        });
       } else {
         attributeCheckBoxForm = new FormControl(false);
       }
       this.attributeCheckBoxForms.push(attributeCheckBoxForm);
     }
   }
 
-  /**
-   * Triggered when a value is selected
-   */
+  get allowZeroValueQuantity(): boolean {
+    return (
+      this.quantityService?.allowZeroValueQuantity(this.attribute) ?? false
+    );
+  }
+
   onSelect(): void {
     const selectedValues = this.configUtilsService.assembleValuesForMultiSelectAttributes(
       this.attributeCheckBoxForms,
       this.attribute
     );
 
     const event: ConfigFormUpdateEvent = {
-      ownerKey: this.ownerKey,
       changedAttribute: {
-        name: this.attribute.name,
+        ...this.attribute,
         values: selectedValues,
-        uiType: this.attribute.uiType,
-        groupId: this.attribute.groupId,
       },
+      ownerKey: this.ownerKey,
+      updateType: Configurator.UpdateType.ATTRIBUTE,
+    };
+
+    this.selectionChange.emit(event);
+  }
+
+  protected onHandleAttributeQuantity(quantity: number): void {
+    this.loading$.next(true);
+
+    const event: ConfigFormUpdateEvent = {
+      changedAttribute: {
+        ...this.attribute,
+        quantity,
+      },
+      ownerKey: this.ownerKey,
+      updateType: Configurator.UpdateType.ATTRIBUTE_QUANTITY,
+    };
+
+    this.selectionChange.emit(event);
+  }
+
+  onChangeValueQuantity(
```

Comment:
There is also `onChangeValueQuantity` in `ConfiguratorAttributeMultiSelectionBundleComponent`, please consider move them to parent.

Reply:
@AlexanderKuschner, could you have a look at it?

Reply:
As discussed: addressed with #12057